functest/create_test.go

```diff
@@ -14,163 +14,157 @@
-	defer cleanup()
-
-	// Because the 'starter' folder has been copied, the stack.yaml file will be in the 'starter'
-	// folder within the temp directory that has been generated for sandboxing purposes, rather than
-	// the usual core temp directory
-	sandbox.ProjectDir = filepath.Join(sandbox.TestDataPath, "starter")
-
-	packageArgs := []string{"stack", "package"}
-	_, err := cmdtest.RunAppsody(sandbox, packageArgs...)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	createArgs := []string{"stack", "create", "testing-stack", "--copy", "dev.local/starter"}
-	_, err = cmdtest.RunAppsody(sandbox, createArgs...)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	exists, err := cmdtest.Exists(filepath.Join(sandbox.ProjectDir, "testing-stack"))
-
-	if err != nil {
-		t.Fatal("Error checking if the stack exists: ", err)
-	}
-	if !exists {
-		t.Fatal("Stack doesn't exist despite appsody stack create executing correctly.")
-	}
-
-}
-
-func TestStackCreateCustomRepo(t *testing.T) {
-
-	sandbox, cleanup := cmdtest.TestSetupWithSandbox(t, true)
-	defer cleanup()
-
-	// Because the 'starter' folder has been copied, the stack.yaml file will be in the 'starter'
-	// folder within the temp directory that has been generated for sandboxing purposes, rather than
-	// the usual core temp directory
-	sandbox.ProjectDir = filepath.Join(sandbox.TestDataPath, "starter")
-
-	packageArgs := []string{"stack", "package"}
-	_, err := cmdtest.RunAppsody(sandbox, packageArgs...)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	devlocalFolder := filepath.Join(sandbox.ConfigDir, "stacks", "dev.local")
-
-	addToRepoArgs := []string{"stack", "add-to-repo", "test-repo", "--release-url", "file://" + devlocalFolder + "/"}
-
-	_, err = cmdtest.RunAppsody(sandbox, addToRepoArgs...)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	testRepoIndex := filepath.Join(devlocalFolder, "test-repo-index.yaml")
-
-	addRepoArgs := []string{"repo", "add", "test-repo", "file://" + testRepoIndex}
-	_, err = cmdtest.RunAppsody(sandbox, addRepoArgs...)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	createArgs := []string{"stack", "create", "testing-stack", "--copy", "test-repo/starter"}
-	_, err = cmdtest.RunAppsody(sandbox, createArgs...)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	exists, err := cmdtest.Exists(filepath.Join(sandbox.ProjectDir, "testing-stack"))
-
-	if err != nil {
-		t.Fatal("Error checking if the stack exists: ", err)
-	}
-	if !exists {
-		t.Fatal("Stack doesn't exist despite appsody stack create executing correctly.")
+func TestFuncCreateValidCases(t *testing.T) {
+	var stackCreateValidTests = []struct {
+		testName   string
+		addToRepo  bool
+		createArgs []string
+		stackName  string
+	}{
+		{"Create with dev.local stack", false, []string{"stack", "create", "testing-stack", "--copy", "dev.local/starter"}, "testing-stack"},
+		{"Create with custom repo stack", true, []string{"stack", "create", "testing-stack", "--copy", "test-repo/starter"}, "testing-stack"},
```

Comment:
Just a question: if all of the cases in this test will be `stack create`'s, is it necessary to include it in each argument array? You could instead do a `args := append([]string{"stack", "create"}, tt.createArgs...)` in the future!